charmhelpers/core/hookenv.py

```diff
@@ -18,7 +18,10 @@
 #  Charm Helpers Developers <juju@lists.ubuntu.com>
 
 import copy
-from distutils.version import LooseVersion
+try:
+    from distutils.version import LooseVersion
+except ImportError:
+    from looseversion import LooseVersion
```

Comment:
shouldn't this be more like:

```python
try:
    from distutils.version import LooseVersion
except ImportError:
    try:
        # reactive charm path, where the venv already has this package
        from looseversion import LooseVersion
    except ImportError:
        # classic charm code path where python packages are installed via apt-get
        apt_install(['python3-looseversion'], fatal=True)
        from looseversion import LooseVersion
```

Reply:
I really don't like this pattern where charmhelpers runs `apt_install`; I'd much rather the parent charm resolved the dependencies and installed python3-looseversion where necessary; (which might be all of them in the future).  It makes unit testing harder when `apt_install()` is at module load time.